ui/src/clockface/components/color_picker/ColorPicker.tsx

```diff
@@ -0,0 +1,192 @@
+// Libraries
+import React, {Component, ChangeEvent} from 'react'
+import _ from 'lodash'
+
+// Components
+import {
+  Button,
+  IconFont,
+  ButtonShape,
+  ComponentStatus,
+} from '@influxdata/clockface'
+import {Input} from 'src/clockface'
+import Swatch from 'src/clockface/components/color_picker/ColorPickerSwatch'
+import Error from 'src/clockface/components/form_layout/FormElementError'
+
+// Constants
+import {colors} from 'src/clockface/constants/colors'
+
+// Utils
+import {validateHexCode} from 'src/configuration/utils/labels'
+
+// Styles
+import 'src/clockface/components/color_picker/ColorPicker.scss'
+
+interface PassedProps {
+  selectedHex: string
+  onSelect: (hex: string, status?: ComponentStatus) => void
+}
+
+interface DefaultProps {
+  maintainInputFocus?: boolean
+  testID?: string
+}
+
+type Props = PassedProps & DefaultProps
+
+interface State {
+  inputValue: string
+  status: string
+}
+
+export default class ColorPicker extends Component<Props, State> {
+  public static defaultProps: DefaultProps = {
+    maintainInputFocus: false,
+    testID: 'color-picker',
+  }
+
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      inputValue: this.props.selectedHex || '',
+      status: null,
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.selectedHex !== this.props.selectedHex) {
+      this.setState({inputValue: this.props.selectedHex})
```

Comment:
is it necessary to keep track of the hex value in state? it seems like this will be creating two sources of truth? could all the places that state instead call the onSelect() prop?